presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/util/RetryDriver.java

```diff
@@ -72,7 +82,8 @@ private RetryDriver()
                 DEFAULT_MAX_RETRY_TIME,
                 Optional.empty(),
                 e -> false,
-                Function.identity());
+                Function.identity(),
+                MoreExecutors.listeningDecorator(Executors.newScheduledThreadPool(DEFAULT_RETRY_THREAD_POOL_SIZE)));
```

Comment:
you need to make sure you have enough threads here. I'd use something like caching thread pool. Also, use naming factory and name the threads. See examples in the code.